File: src/main/scala/com/digitalasset/testing/logging/Dump.scala
/*
 * Copyright 2019 Digital Asset (Switzerland) GmbH and/or its affiliates
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package com.digitalasset.testing.logging
import grizzled.slf4j.Logger

object Dump {
  def dump(logger: String, event: => LogEvent): Unit = {
    val log = Logger(logger)
    if (log.isDebugEnabled) {
      val contextString = getContextString
      if (contextString != null)
        Logger(logger).logger
          .debug(event.consolePrettyMsg, event: Any, getContextString: Any)
      else
        Logger[Dump.type]
          .debug("Skipping wire dump, the context is not provided")
    }
  }

  private var context: String = _
  def setContextString(context: String): Unit = synchronized {
    this.context = context
  }
  def getContextString: String = synchronized { context }
  def clearContextString(): Unit = synchronized { context = null }
}

File: src/main/java/com/digitalasset/testing/logging/Dump.java
/*
 * Copyright 2019 Digital Asset (Switzerland) GmbH and/or its affiliates
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package com.digitalasset.testing.logging;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Dump {
  private static volatile String context;

  public static void dump(String logger, LogEvent event) {
    Logger log = LoggerFactory.getLogger(logger);
    if (log.isDebugEnabled()) {
      String contextString = context;
      if (contextString != null) {
        log.debug(event.consolePrettyMsg(), event, contextString);
      } else {
        log.debug("Skipping wire dump, the context is not provided");
      }
    }
  }

  public static synchronized void setContextString(String c) {
    context = c;
  }

  public static synchronized void clearContextString() {
    context = null;
  }
}
